overlord/aspectstate/aspectstate.go

```diff
@@ -85,6 +85,51 @@ func GetAspect(databag aspects.DataBag, account, bundleName, aspect, field strin
 	return nil
 }
 
+type Filters struct {
+	params map[string]Filter
+}
+
+type Filter struct {
+	value    string
+	optional bool
+}
+
+func QueryAspect(databag aspects.DataBag, account, bundleName, aspect, request, query string) ([]interface{}, error) {
+	patt := map[string]interface{}{
+		"asp": []map[string]string{
+			{"name": "byname", "path": "snaps.{name}"},
```

Comment:
this is strange because name would be a request (so a dotted name) not an abstract name for the query, so it would be snaps.{name} again, that's also why we then need to support nesting